### DISCUSSION TITLE
[point] 스타일링 적용시 width, height 강제 단점 공유

### DISCUSSION THREAD

**Discussion by July249** [beyondDevelops/AnimalTalk#67] [Show and tell]
스타일링을 적용함에 있어 너비와 높이를 스타일에 강제하는 것은 좋지 않습니다. 그 이유는 내부 콘텐츠가 차지할 수 있는 공간의 크기를 제한하고 이로 인해 overflow 현상이 발생할 여지가 존재하기 때문입니다.

참고하시면 좋을 mdn 공식 문서의 자료 입니다.

[ 참고 자료 ]
https://developer.mozilla.org/ko/docs/Learn/CSS/Building_blocks/Sizing_items_in_CSS#%ED%8A%B9%EC%A0%95_%ED%81%AC%EA%B8%B0_%EC%84%A4%EC%A0%95

**Comment by GreattitJY** [comment]
해당 경우 텍스트나 이미지와 같은 레이아웃 요소가 동적으로 들어가서 그러는 걸로 이해하고 있습니다. 그런데 헤더와 같은 요소는 안의 콘텐츠가 이미 정해져 있고 앱 어디에서나 같은 크기를 유지해야 하는데 동적으로 두기는 위험할 거 같습니다... 물론 height 대신 padding을 주는 것도 괜찮은 방법일 거 같습니다..!

**Reply by July249** [reply]
헤더 작업 때 참여하지 못해 misunderstanding 이 있었던 것 같습니다. 그런 고민이 있었군요! 하지만 헤더 내부에 콘텐츠(이미지, 텍스트, 버튼 등)가 있기 때문에 내부 콘텐츠를 이용해서 높이를 맞추는 편이 조금 좋을 것 같습니다. ㅎㅎ

전반적인 내부 콘텐츠의 너비는 전체 레이아웃에 width를 지정하여 주면 넘어가지 않는 것 같습니다.

재영님이 말씀해주신 것과 같이 오히려 내부 콘텐츠(대표적으로 이미지)의 크기가 너무 다양해서 받아오는 데이터에 따라 콘텐츠가 레이아웃에 미치는 유동성이 전반적인 레이아웃을 해치는 경우라면 너비와 높이를 지정하여 주는 것이 좋은 방법인 것 같습니다. ^^
